overlord/devicestate/handlers.go

```diff
@@ -57,6 +57,50 @@ func (m *DeviceManager) doMarkSeeded(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+func isSameAssertsRevision(err error) bool {
+	if e, ok := err.(*asserts.RevisionError); ok {
+		if e.Used == e.Current {
+			return true
+		}
+	}
+	return false
+}
+
+func (m *DeviceManager) doSetModel(t *state.Task, _ *tomb.Tomb) error {
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+
+	var modelass []byte
+	if err := t.Get("new-model", &modelass); err != nil {
+		return err
+	}
+
+	ass, err := asserts.Decode(modelass)
+	if err != nil {
+		return err
+	}
+
+	model, ok := ass.(*asserts.Model)
+	if !ok {
+		return fmt.Errorf("new-model is not a model assertion but: %s", ass.Type().Name)
```

Comment:
this is an internal error at this point, no? we should fail in Remodel if that was not the case?

Reply:
Yes, I will update the error.